nibabel/streamlines/tck.py

```diff
@@ -0,0 +1,450 @@
+        header.update(self.header)
+
+        # Keep counts for correcting incoherent fields or warn.
+        nb_streamlines = 0
+
+        with Opener(fileobj, mode="wb") as f:
+            # Keep track of the beginning of the header.
+            beginning = f.tell()
+
+            # Write temporary header that we will update at the end
+            self._write_header(f, header)
+
+            try:
+                first_item = next(iter(self.tractogram))
+            except StopIteration:
+                # Empty tractogram
+                header[Field.NB_STREAMLINES] = 0
+                self._finalize_header(f, header, offset=beginning)
+
+                # Add the EOF_DELIMITER.
+                f.write(asbytes(self.EOF_DELIMITER.tostring()))
+                return
+
+            data_for_streamline = first_item.data_for_streamline
+            if len(data_for_streamline) > 0:
+                keys = ", ".join(data_for_streamline.keys())
+                msg = ("TCK format does not support saving additional data"
+                       " alongside streamlines. Dropping: {}".format(keys))
+                warnings.warn(msg, DataWarning)
+
+            data_for_points = first_item.data_for_points
+            if len(data_for_points) > 0:
+                keys = ", ".join(data_for_points.keys())
+                msg = ("TCK format does not support saving additional data"
+                       " alongside points. Dropping: {}".format(keys))
+                warnings.warn(msg, DataWarning)
+
+            # Make sure streamlines are in rasmm.
+            tractogram = self.tractogram.to_world(lazy=True)
+
+            for t in tractogram:
+                s = t.streamline.astype(dtype)
+                data = np.r_[s, self.FIBER_DELIMITER]
+                f.write(asbytes(data.tostring()))
+                nb_streamlines += 1
+
+            header[Field.NB_STREAMLINES] = nb_streamlines
+
+            # Add the EOF_DELIMITER.
+            f.write(asbytes(self.EOF_DELIMITER.tostring()))
+            self._finalize_header(f, header, offset=beginning)
+
+    @staticmethod
+    def _write_header(fileobj, header):
+        """ Write TCK header to file-like object.
+
+        Parameters
+        ----------
+        fileobj : file-like object
+            An open binary file object pointing to TCK file (and ready to
+            write at the beginning of the TCK header).
+        """
+        # Fields to exclude
+        exclude = [Field.MAGIC_NUMBER,  # Handled separately.
+                   Field.NB_STREAMLINES,  # Handled separately.
+                   Field.ENDIANNESS,  # Handled separately.
+                   Field.VOXEL_TO_RASMM,  # Streamlines are always in RAS+ mm.
+                   "count", "datatype", "file"]  # Fields being replaced.
+
+        lines = []
+        lines.append(header[Field.MAGIC_NUMBER])
+        lines.append("count: {0:010}".format(header[Field.NB_STREAMLINES]))
```

Comment:
Actually it would need more than 10 **Billions** streamlines. If that is an issuewe can simply bump this number up.